src/components/SocialShareButton.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { FC } from 'react';
+import { StyleSheet, TouchableOpacity } from 'react-native';
+import Share, { Options } from 'react-native-share';
+
+import { Image } from 'app/components';
+
+import { captureException } from '../../error';
+
+interface Props {
+  source: number;
+  shareOptions: Options & { social: Share.Social };
+}
+
+export const SocialShareButton: FC<Props> = ({ source, shareOptions }) => {
+  const share = async () => {
+    try {
+      await Share.shareSingle(shareOptions);
+    } catch ({ error }) {
+      const errorMsg = `SocialShareButton ${shareOptions.social} error: ${error}`;
+      captureException(errorMsg);
```

Comment:
I like that 👍 but I have a few concerns:
- are you sure it should be `catch ({ error })` instead of `catch (error)`?
- I think it'd be safer to covert error object to string with `JSON.stringify()` first before putting it into `errorMsg`
- the `share` function name says not much, to be honest. I had to check in the documentation to find out what's the exact behavior of `Share.shareSingle()` method. Maybe the name of `openShareDialog` would be better?

Reply:
Added JSON.stringify(). It doesn't open share dialog, changed to `addPostToSocialMediaApp`